test/expr/unary/keypath/keypath.swift

```diff
@@ -1178,3 +1187,50 @@ func sr14499() {
     // expected-note@-1 2 {{cast 'Any' to 'AnyObject' or use 'as!' to force downcast to a more specific type to access members}}
   }
 }
+
+func keypath_function_transitive_conversions() {
+      class Base {
+          var derived: Derived { Derived() }
+          var base: Base { Base() }
+          var int: Int { 0 }
+      }
+
+      class Derived: Base {
+          override var derived: Derived { Derived() }
+          override var base: Base { Base() }
+      }
+
+      struct S {
+          var base: Base { Base() }
+          var derived: Derived { Derived() }
+      }
+
+      let _: (Base) -> Base = \Base.base
+      let _: (Base) -> Base = \Derived.base
+      let _: (Base) -> Base? = \Base?.self
+      let _: (Base) -> Base? = \Base?.self?.base
+      // FIXME: This error text is bogus due to KeyPath base covariance.
+      let _: (Base?) -> Base = \Base.base // expected-error {{value of optional type 'Optional<Base>' must be unwrapped to refer to member 'base' of wrapped base type 'Base'}} expected-note {{use unwrapped type 'Base' as key path root}} {{33-37=Base}}
```

Comment:
Note: we get the same error without the function conversion, e.g.,
```
let _: KeyPath<Base?, Base> = \Base.base
```

Filed [here](https://bugs.swift.org/browse/SR-15292).